Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,14 +41,58 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            const pypirc = fs.readFileSync(tl.getVariable("PYPIRC_PATH"), 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let dotRepos: string[] = [];
+
+            for (let entry of newEndpointsToAdd) {
+
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
+
+                if (entry.packageSource.feedName in fileContent){
+                    tl.debug(tl.loc("DuplicateRegistry", entry.packageSource.feedName));
```

Comment:
Can we make this a warning instead of a debug ?

Reply:
Changed this!